CHANGELOG.md

```diff
@@ -1,5 +1,34 @@
 # Changelog
 
+## grantami-recordlists 1.2.1, 2024-05-07
+
+### Changed
```

Comment:
Is 'Changed' the right word here? Maybe 'Dependencies'? I'm not sure what the towncrier template uses, but maybe we should standardize on that

Reply:
Yea, let's go with Dependencies. At least then it's clear that the only meaningful fix is that change

Reply:
Any thoughts on whether the other cherry picked PRs ought to be listed there too? I did not include them, based on what was in the changelog for previous versions